src/spring/azext_spring/_app_factory.py

```diff
@@ -24,8 +24,21 @@ def _format_properties(self, **kwargs):
         kwargs['vnet_addons'] = self._load_vnet_addons(**kwargs)
         kwargs['ingress_settings'] = self._load_ingress_settings(**kwargs)
         kwargs['secrets'] = self._load_secrets_config(**kwargs)
+        kwargs['test_endpoint_auth_state'] = self._get_test_endpoint_auth_state(**kwargs)
         kwargs['addon_configs'] = self._load_addon_configs(**kwargs)
-        return models.AppResourceProperties(**kwargs)
+        properties = models.AppResourceProperties(**kwargs)
+
+        # When execute app update without parameter 'disable_test_endpoint_auth', use 'None' instead of 'Enabled' for property test_endpoint_auth_state
+        if kwargs['disable_test_endpoint_auth'] is None:
+            properties.test_endpoint_auth_state = None
+
+        return properties
```

Comment:
Ignore these changes

Reply:
The class 'models.AppResourceProperties' has a default value 'Enabled' for property 'test_endpoint_auth_state ', this checking is used to keep the property none if the 'app update' command does not include the parameter 'disable_test_endpoint_auth', otherwise the app's test endpoint will be re-set to enabled.